### PR TITLE
Squash logged errors from failed SQL rollbacks

### PR DESCRIPTION
**Pull request by mheon**
Currently we unconditionally roll back transactions after error, even if a commit has already been attempted. Commit is guaranteed to end a transaction, though, whether by successfully committing or by rolling back if that fails. As such, we attempt a double rollback if a transaction fails at commit (for example, for a constraint violation), which doesn't error but does log angry warning messages. Ensure we don't try rolling back after commit runs to prevent this.

### CODE REVIEW

**Comment by mheon** [comment]
This doesn't matter much now that we're on bolt, but the errors logged during our unit tests were annoying me

File: libpod/sql_state.go [hunk 1]
```diff
@@ -135,6 +137,8 @@ func (s *SQLState) Refresh() (err error) {
 		return errors.Wrapf(err, "error refreshing database state")
 	}
 
+	committed = true
```
Comment (TomSweeneyRedHat) [inline libpod/sql_state.go]: Should this be after the next if statement?  If tx.Commit has an error, will it be commited?  Same comment through out.

Comment (TomSweeneyRedHat) [reply libpod/sql_state.go]: OK, good to know, thanks!

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**